src/Common/src/TypeSystem/IL/Stubs/PInvokeMarshallingILEmitter.cs

```diff
@@ -510,22 +510,47 @@ public MethodIL EmitIL()
 
             MethodSignature nativeSig = new MethodSignature(
                 targetMethodSignature.Flags, 0, nativeReturnType, nativeParameterTypes);
-            MethodDesc nativeMethod =
-                new PInvokeTargetNativeMethod(_targetMethod.OwningType, nativeSig, _importMetadata);
 
-            // Call the native method
-            callsiteSetupCodeStream.Emit(ILOpcode.call, _emitter.NewToken(nativeMethod));
 
+            bool useLazyResolution;
+            string importModule = _importMetadata.Module;
+
+            // Determine whether this call should be made through a lazy resolution or a static reference
+            // Eventually, this should be controlled by a custom attribute (or an extension to the metadata format).
+            if (importModule == "[MRT]" || importModule == "*")
+                useLazyResolution = false;
+            else if (_targetMethod.Context.Target.IsWindows)
+                useLazyResolution = !importModule.StartsWith("api-ms-win-core-libraryloader-l1-2-0");
```

Comment:
Note to self: fix this before merging. There's not much to test if this is just `api-ms-win-`